internal/manifests/collector/config_replace.go

```diff
@@ -33,12 +34,57 @@ type Config struct {
 }
 
 func ReplaceConfig(instance v1alpha1.AmazonCloudWatchAgent) (string, error) {
+	// Parse the original configuration from instance.Spec.Config
 	config, err := adapters.ConfigFromJSONString(instance.Spec.Config)
 	if err != nil {
 		return "", err
 	}
 
-	out, err := json.Marshal(config)
+	conf := confmap.NewFromStringMap(config)
+
+	prometheusFilePath := conf.Get("logs::metrics_collected::prometheus::prometheus_config_path")
+	if prometheusFilePath == nil {
+		prometheusFilePath = "/etc/prometheusconfig/prometheus.yaml"
+	}
+	if conf.IsSet("logs::metrics_collected::prometheus") && !instance.Spec.Prometheus.IsEmpty() {
+		prometheusConfig := confmap.NewFromStringMap(map[string]interface{}{
```

Comment:
nit: this could be a global variable called `DefaultPrometheusPath` and can be reused. We don't need to create a new map everytime.